sw/vendor/patches/mundane/build_with_bazel.patch

```diff
@@ -0,0 +1,2245 @@
++    pub fn SHA512_Final(out: *mut u8, sha: *mut SHA512_CTX) -> ::std::os::raw::c_int;
++}
++#[repr(C)]
++#[derive(Debug, Copy, Clone)]
++pub struct sha512_state_st {
++    pub h: [u64; 8usize],
++    pub Nl: u64,
++    pub Nh: u64,
++    pub p: [u8; 128usize],
++    pub num: ::std::os::raw::c_uint,
++    pub md_len: ::std::os::raw::c_uint,
++}
++#[test]
++fn bindgen_test_layout_sha512_state_st() {
++    assert_eq!(
++        ::std::mem::size_of::<sha512_state_st>(),
++        216usize,
++        concat!("Size of: ", stringify!(sha512_state_st))
++    );
++    assert_eq!(
++        ::std::mem::align_of::<sha512_state_st>(),
++        8usize,
++        concat!("Alignment of ", stringify!(sha512_state_st))
++    );
++    assert_eq!(
++        unsafe { &(*(::std::ptr::null::<sha512_state_st>())).h as *const _ as usize },
++        0usize,
++        concat!("Offset of field: ", stringify!(sha512_state_st), "::", stringify!(h))
++    );
++    assert_eq!(
++        unsafe { &(*(::std::ptr::null::<sha512_state_st>())).Nl as *const _ as usize },
++        64usize,
++        concat!("Offset of field: ", stringify!(sha512_state_st), "::", stringify!(Nl))
++    );
++    assert_eq!(
++        unsafe { &(*(::std::ptr::null::<sha512_state_st>())).Nh as *const _ as usize },
++        72usize,
++        concat!("Offset of field: ", stringify!(sha512_state_st), "::", stringify!(Nh))
++    );
++    assert_eq!(
++        unsafe { &(*(::std::ptr::null::<sha512_state_st>())).p as *const _ as usize },
++        80usize,
++        concat!("Offset of field: ", stringify!(sha512_state_st), "::", stringify!(p))
++    );
++    assert_eq!(
++        unsafe { &(*(::std::ptr::null::<sha512_state_st>())).num as *const _ as usize },
++        208usize,
++        concat!("Offset of field: ", stringify!(sha512_state_st), "::", stringify!(num))
++    );
++    assert_eq!(
++        unsafe { &(*(::std::ptr::null::<sha512_state_st>())).md_len as *const _ as usize },
++        212usize,
++        concat!("Offset of field: ", stringify!(sha512_state_st), "::", stringify!(md_len))
++    );
++}
+diff --git a/src/boringssl/mod.rs b/src/boringssl/mod.rs
+index 18e69b9..272d990 100644
+--- a/src/boringssl/mod.rs
++++ b/src/boringssl/mod.rs
+@@ -74,8 +74,13 @@
+ // NOTE(joshlf): It's important to define this module before the abort module,
+ // or else all of the assertions that are auto-generated by bindgen would result
+ // in compilation errors.
+-#[path = "../../boringssl/boringssl.rs"]
++#[cfg(not(feature = "bazel_build"))]
++#[path = "../../boringssl/boringssl_with_renames.rs"]
+ mod ffi;
++#[cfg(feature = "bazel_build")]
++#[path = "../../boringssl/boringssl_without_renames.rs"]
++mod ffi;
++
+ #[macro_use]
+ mod abort;
+ #[macro_use]
+diff --git a/src/lib.rs b/src/lib.rs
+index 36c5806..1b22a17 100644
+--- a/src/lib.rs
++++ b/src/lib.rs
+@@ -58,6 +58,7 @@ mod macros;
+ 
+ // Forbid unsafe code except in the boringssl module.
+ #[allow(unsafe_code)]
++#[allow(deref_nullptr)]
```

Comment:
Uhhhh im gonna assume there's a good reason for this.

Reply:
Yes.  The bindgen created bindings have a number of tests that check struct offsets.  I believe the mundane library is meant to work with older editions of the rust language (2015), prior to the addition of `offset_of`.  As such, the struct offset checks have the following form:

```
    assert_eq!(
        unsafe { &(*(::std::ptr::null::<bignum_st>())).d as *const _ as usize },
        0usize,
        concat!("Offset of field: ", stringify!(bignum_st), "::", stringify!(d))
    );
```

When building with the bazel rust rules, the compiler complains about this.